### DISCUSSION TITLE
Voltage in UI

### DISCUSSION THREAD

**Discussion by Mythris29** [sanni/cartreader#484] [Ideas & Suggestions]
I think it would be nice to have the voltage settings in the title of the dumping module somewhere so I don't have to keep going back to the wiki :) 

Something like going into the Game Boy menu Game Boy Advance would have a little 3V next to it to remind you to switch to that mode.

**Comment by Icesythe7** [comment]
I would agree with that except for the fact that unless you strictly use original carts most bootlegs/aftermarket stuff uses different voltages, ie gb/gbc games all use 5v gba uses 3v however all the bootleg pokemon games etc for gb/gbc all use 3v flash chips and ur gb over voltages them the entire time lol

**Reply by Mythris29** [reply]
That's fair but I think that at least starting with the "normal" voltage is probably a good idea then you can always play with others if the cart doesn't dump correctly.

And to be clear I'm not proposing the cart reader automatically sets the voltage I'm just wanting a reminder of what mode I should have my switch for 3v -5v set to for each console.

**Reply by Icesythe7** [reply]
you can change that here for gb in like 3 seconds if you want I think sanni also changed some already, simply add whatever text you want to whichever menu item you want and reupload
https://github.com/sanni/cartreader/blob/master/Cart_Reader/GB.ino#L18

**Reply by Mythris29** [reply]
Oh cool yeah that's helpful and it does look like it's been added to GB advance already. Since the 3v carts seem a lot more rare than 5v I think that approach makes sense to maybe just put a 3V next to those that need that.

**Reply by Icesythe7** [reply]
I think you are misunderstanding the voltages, it is not that "3v carts are rarer" it is what did the system use? ie all gameboys and gameboy colors use 5v only gameboy advance use 3v, generally the older the system the older the chips and the older chips use 5v as technology advanced the newer chips use 3v, if you look at a gameboy cart and a gamboy advance cart you will see the bottom of the advance cart has a notch cut out this misses the 5v switch in the gba making it operate in 3v mode, if you put in a gb/gbc cart it hits the switch and puts it in 5v mode, you can see the switch aswell on the side if you look inside the gba cart slot. :)

**Reply by Mythris29** [reply]
Yeah, no I understand that the systems had a standard voltage they use I thought you were saying that some repo carts do strange things and different voltages than what the original system used. I misunderstood that.

As for 3V being rarer I just meant there are only 2 currently supported systems that use that voltage (at least that I could find in the wiki). The N64 and GBA (like you said the newer ones) so it makes sense to me to only denote the 3V systems in the menu since all the rest are 5V. That's all I meant by that comment.

**Reply by Ancyker** [reply]
I would agree with that if not for the fact sending 3V to a 5V cart would generally just result in it not working while sending 5V to a 3V cart can damage it. I was actually looking into making voltage auto-select depending on the cart type selected. There is no more harm in inserting a 3V cart for a 5V system into a 5V cart reader than there is using that same cart with the actual system it's for. Personally, I'd much rather risk damaging a flash cart, which is usually relatively inexpensive, vs an original cart which could be worth $100+, especially when the damage done would be caused by the system it's for anyway.
